includes/functions.inc.php

```diff
@@ -322,13 +322,14 @@ function arrayClean(&$array) {
 
 /**
  * Recursively strip HTML from a (multidimensional) array.
+ * Assumes entity decoding to UTF-8 (primarily for OAI-PMH)
  * @param $values array
  * @return array the cleansed array
  */
-function stripAssocArray($values) {
+function stripAssocArray($values, $useClientCharset = false) {
 	foreach ($values as $key => $value) {
 		if (is_scalar($value)) {
-			$values[$key] = strip_tags($values[$key]);
+			$values[$key] = html_entity_decode(strip_tags($values[$key]), ENT_QUOTES, $useClientCharset ? Config::getVar('i18n', 'client_charset') : 'UTF-8');
```

Comment:
@ctgraham, we also have String::html2utf as an alternative to PHP's built-in html_entity_decode.